CAIPs/caip-25.md

```diff
@@ -61,61 +61,175 @@ If a connection is rejected, the wallet MAY respond with a generic error or sile
 
 #### Request
 
+```typescript
+interface CAIP25JsonRpcRequest {
+  id: number;
+  jsonrpc: "2.0";
+  method: "wallet_createSession";
+  params: {
+    scopes: {
+      [scopeKey: string]: {
+        chains?: string[];
+        methods: string[];
+        notifications: string[];
+      };
+    };
+    properties?: {
+      [propertyKey: string]: any;
+    };
+  };
+}
+```
+
+The `scopes` object MUST contain one or more scopeObjects.
+
+The `properties` object MAY be included for global session metadata.
+
+### Response
+
+#### Success
+
+```typescript
+interface CAIP25JsonRpcRequest {
+  id: number;
+  jsonrpc: "2.0";
+  result: {
+    scopes: {
+      [scopeKey: string]: {
+        chains?: string[];
+        accounts: string[];
+        methods: string[];
+        notifications: string[];
+        capabilities?: {
+          [capabilityKey: string]: any;
+        };
+      };
+    };
+    properties?: {
+      [propertyKey: string]: any;
+    };
+  };
+}
+```
+
+Each entry within `scopes` object MAY contain `accounts` and `capabilities` as part of its object for success response.
+
+#### Error Codes
+
+The wallet MAY return generic or specific error messages depending on trust. Trusted responses may include codes like:
+
+- `5000`: Unknown error
+- `5001`: User disapproved requested methods
+- `5002`: User disapproved requested notifications
+- `5100-5102`: Unsupported chains, methods, or notifications
+- `5201-5302`: Malformed requests
+
+## Examples
+
+**Example 1**
+
+For request, we define a very simple scope for 10 EVM chains with the exact same scope.
+
 ```jsonc
+// JSON-RPC REQUEST
 {
   "id": 1,
   "jsonrpc": "2.0",
   "method": "wallet_createSession",
   "params": {
     "scopes": {
-      "wallet": {
-        "methods": [
-          "wallet_revokeSession",
-          "wallet_getSession",
-          "wallet_authenticate",
-          "wallet_pay"
+      "eip155": {
+        "chains": [
+          "1",
+          "10",
+          "130",
+          "324",
+          "2741",
+          "8453",
+          "42161",
+          "59144",
+          "534352",
+          "747474"
         ],
-        "notifications": ["wallet_sessionChanged"]
-      },
-      "wallet:eip155": {
```

Comment:
sorry. what was our decision around a scope for parking non-chain-specific methods? Are we just saying that they must be parked inside a chain specific scope?